lib/dry/configurable.rb

```diff
@@ -122,6 +126,10 @@ def _reader_attributes
 
     private
 
+    def extract_preprocessor(value)
+      value.respond_to?(:call) ? [nil, value] : [value, nil]
```

Comment:
I know it doesn't explain much but I will improve the code probably in the `ArgumentParser` class to handle all this logic